### ISSUE TITLE
 Pull access denied for dokku/herokuish

### ISSUE THREAD

**Issue by justintemps** [dokku/dokku#6401]
### Description of problem

I'm trying to upgrade from v0.21.0 to 0.32.0. After running the steps indicated in the [Upgrade Page](https://dokku.com/docs/getting-started/upgrading/), I get the following error when running `dokku ps:rebuild --all`:

```shell
 !     Running rebuild in serial mode
 -----> Running rebuild against app weso-dashboard-backend-staging
-----> Cleaning up...
-----> Building weso-dashboard-backend-staging from herokuish
 !     Deprecated: please upgrade plugin to use 'pre-build' plugin trigger instead of pre-build-buildpack
-----> Adding app specific deployment-keys to build environment ...
-----> Adding host-keys to build environment ...
       Adding shared keys
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Adding shared deployment-keys to build environment ...
Unable to find image 'dokku/herokuish:latest' locally
docker: Error response from daemon: pull access denied for dokku/herokuish, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
 !     Removing invalid image tag dokku/weso-dashboard-backend-staging:latest
 !     App build failed
 !     Error running rebuild against app weso-dashboard-backend-staging: exit status 1
```

### Steps to reproduce

Attempt to upgrade a dokku deploy from v0.21.0 to 0.32.0

### dokku report $APP_NAME

```shell
-----> uname: Linux weso-dashboard-upgrade-1659029142806-c-4-8gib-fra1-01 5.4.0-122-generic #138-Ubuntu SMP Wed Jun 22 15:00:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:           7957         547        3699           1        3710        7101
       Swap:             0           0           0
-----> docker version:
       Client: Docker Engine - Community
        Version:           24.0.7
        API version:       1.43
        Go version:        go1.20.10
        Git commit:        afdd53b
        Built:             Thu Oct 26 09:08:01 2023
        OS/Arch:           linux/amd64
        Context:           default

       Server: Docker Engine - Community
        Engine:
         Version:          24.0.7
         API version:      1.43 (minimum version 1.12)
         Go version:       go1.20.10
         Git commit:       311b9ff
         Built:            Thu Oct 26 09:08:01 2023
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.6.26
         GitCommit:        3dd1e886e55dd695541fdcd67420c2888645a495
        runc:
         Version:          1.1.10
         GitCommit:        v1.1.10-0-g18a0cb0
        docker-init:
         Version:          0.19.0
         GitCommit:        de40ad0
-----> docker daemon info:
       Client: Docker Engine - Community
        Version:    24.0.7
        Context:    default
        Debug Mode: true
        Plugins:
         buildx: Docker Buildx (Docker Inc.)
           Version:  v0.11.2
           Path:     /usr/libexec/docker/cli-plugins/docker-buildx
         compose: Docker Compose (Docker Inc.)
           Version:  v2.21.0
           Path:     /usr/libexec/docker/cli-plugins/docker-compose

       Server:
        Containers: 66
         Running: 3
         Paused: 0
         Stopped: 63
        Images: 89
        Server Version: 24.0.7
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Using metacopy: false
         Native Overlay Diff: true
         userxattr: false
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Cgroup Version: 1
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: io.containerd.runc.v2 runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 3dd1e886e55dd695541fdcd67420c2888645a495
        runc version: v1.1.10-0-g18a0cb0
        init version: de40ad0
        Security Options:
         apparmor
         seccomp
          Profile: builtin
        Kernel Version: 5.4.0-122-generic
        Operating System: Ubuntu 20.04.6 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 4
        Total Memory: 7.771GiB
        Name: weso-dashboard-upgrade-1659029142806-c-4-8gib-fra1-01
        ID: dd8747f4-74d6-4c59-88ae-387fd8b887a8
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
         File Descriptors: 42
         Goroutines: 49
         System Time: 2024-01-03T10:41:14.686811294Z
         EventsListeners: 1
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> herokuish version:
       herokuish: v0.7.2
       buildpacks:
         heroku-buildpack-multi     v1.2.0
         heroku-buildpack-ruby      v259
         heroku-buildpack-nodejs    v224
         heroku-buildpack-clojure   v91
         heroku-buildpack-python    v237
         heroku-buildpack-java      v73
         heroku-buildpack-gradle    v39
         heroku-buildpack-scala     v96
         heroku-buildpack-play      v26
         heroku-buildpack-php       v239
         heroku-buildpack-go        v181
         heroku-buildpack-nginx     v24
         buildpack-null             v3
-----> dokku version: dokku version 0.32.3
-----> dokku-event-listener version: 0.15.0build+5268732
-----> dokku-update version: dokku-update 0.7.2
-----> docker-container-healthchecker version: 0.7.2
-----> docker-image-labeler version: 0.6.1build+c6e15a9
-----> git version: git version 2.25.1
-----> lambda-builder version:        0.5.0
-----> netrc version: 0.7.1build+677c033
 !     pack binary is not available
-----> plugn version: plugn: 0.13.0build+fd5297a
-----> sigil version: 0.10.1build+e443be0
-----> sshcommand version: sshcommand 0.17.1
-----> dokku plugins:
         00_dokku-standard    0.32.3 enabled    dokku core standard plugin
         20_events            0.32.3 enabled    dokku core events logging plugin
         app-json             0.32.3 enabled    dokku core app-json plugin
         apps                 0.32.3 enabled    dokku core apps plugin
         builder              0.32.3 enabled    dokku core builder plugin
         builder-dockerfile   0.32.3 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.32.3 enabled    dokku core builder-herokuish plugin
         builder-lambda       0.32.3 enabled    dokku core builder-lambda plugin
         builder-nixpacks     0.32.3 enabled    dokku core builder-nixpacks plugin
         builder-null         0.32.3 enabled    dokku core builder-null plugin
         builder-pack         0.32.3 enabled    dokku core builder-pack plugin
         buildpacks           0.32.3 enabled    dokku core buildpacks plugin
         caddy-vhosts         0.32.3 enabled    dokku core caddy-vhosts plugin
         certs                0.32.3 enabled    dokku core certificate management plugin
         checks               0.32.3 enabled    dokku core checks plugin
         common               0.32.3 enabled    dokku core common plugin
         config               0.32.3 enabled    dokku core config plugin
         cron                 0.32.3 enabled    dokku core cron plugin
         deployment-keys      0.2.0 enabled    Manage SSH deployment keys that should get injected into your containers on-build
         docker-options       0.32.3 enabled    dokku core docker-options plugin
         domains              0.32.3 enabled    dokku core domains plugin
         enter                0.32.3 enabled    dokku core enter plugin
         git                  0.32.3 enabled    dokku core git plugin
         haproxy-vhosts       0.32.3 enabled    dokku core haproxy-vhosts plugin
         hostkeys-plugin      0.2.1 enabled    Manage hostkeys (.ssh/known_hosts) in your container environment
         logs                 0.32.3 enabled    dokku core logs plugin
         mongo                1.21.0 enabled    dokku mongo service plugin
         network              0.32.3 enabled    dokku core network plugin
         nginx-vhosts         0.32.3 enabled    dokku core nginx-vhosts plugin
         openresty-vhosts     0.32.3 enabled    dokku core openresty-vhosts plugin
         plugin               0.32.3 enabled    dokku core plugin plugin
         ports                0.32.3 enabled    dokku core ports plugin
         proxy                0.32.3 enabled    dokku core proxy plugin
         ps                   0.32.3 enabled    dokku core ps plugin
         registry             0.32.3 enabled    dokku core registry plugin
         repo                 0.32.3 enabled    dokku core repo plugin
         resource             0.32.3 enabled    dokku core resource plugin
         run                  0.32.3 enabled    dokku core run plugin
         scheduler            0.32.3 enabled    dokku core scheduler plugin
         scheduler-docker-local 0.32.3 enabled    dokku core scheduler-docker-local plugin
         scheduler-null       0.32.3 enabled    dokku core scheduler-null plugin
         shell                0.32.3 enabled    dokku core shell plugin
         ssh-keys             0.32.3 enabled    dokku core ssh-keys plugin
         storage              0.32.3 enabled    dokku core storage plugin
         trace                0.32.3 enabled    dokku core trace plugin
         traefik-vhosts       0.32.3 enabled    dokku core traefik-vhosts plugin
=====> weso-dashboard-backend-staging app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:
=====> weso-dashboard-backend-staging app information
       App created at:                1704275860
       App deploy source:
       App deploy source metadata:
       App dir:                       /home/dokku/weso-dashboard-backend-staging
       App locked:                    false
=====> weso-dashboard-backend-staging builder information
       Builder build dir:
       Builder computed build dir:
       Builder computed selected:
       Builder global build dir:
       Builder global selected:
       Builder selected:
=====> weso-dashboard-backend-staging builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile
       Builder dockerfile global dockerfile path: Dockerfile
       Builder dockerfile dockerfile path:
=====> weso-dashboard-backend-staging builder-herokuish information
       Builder herokuish computed allowed: true
       Builder herokuish global allowed: true
       Builder herokuish allowed:
=====> weso-dashboard-backend-staging builder-lambda information
       Builder lambda computed lambdayml path: lambda.yml
       Builder lambda global lambdayml path: lambda.yml
       Builder lambda lambdayml path:
=====> weso-dashboard-backend-staging builder-nixpacks information
       Builder nixpacks computed nixpackstoml path: nixpacks.toml
       Builder nixpacks global nixpackstoml path: nixpacks.toml
       Builder nixpacks nixpackstoml path:
       Builder nixpacks computed no cache: false
       Builder nixpacks global no cache: false
       Builder nixpacks no cache:
=====> weso-dashboard-backend-staging builder-pack information
       Builder pack computed projecttoml path: project.toml
       Builder pack global projecttoml path: project.toml
       Builder pack projecttoml path:
=====> weso-dashboard-backend-staging buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-22
       Buildpacks global stack:
       Buildpacks list:
       Buildpacks stack:
=====> weso-dashboard-backend-staging caddy information
       Caddy image:                   lucaslorentz/caddy-docker-proxy:2.8
       Caddy letsencrypt email:
       Caddy letsencrypt server:      https://acme-v02.api.letsencrypt.org/directory
       Caddy log level:               ERROR
       Caddy polling interval:        5s
       Caddy tls internal:            false
=====> weso-dashboard-backend-staging ssl information
       Ssl dir:                       /home/dokku/weso-dashboard-backend-staging/tls
       Ssl enabled:                   false
       Ssl hostnames:
       Ssl expires at:
       Ssl issuer:
       Ssl starts at:
       Ssl subject:
       Ssl verified:
=====> weso-dashboard-backend-staging checks information
       Checks disabled list:          none
       Checks skipped list:           none
       Checks computed wait to retire: 60
       Checks global wait to retire:  60
       Checks wait to retire:
=====> weso-dashboard-backend-staging docker options information
       Docker options build:          --link dokku.mongo.weso-staging:dokku-mongo-weso-staging
       Docker options deploy:         --link dokku.mongo.weso-staging:dokku-mongo-weso-staging --restart=on-failure:10
       Docker options run:            --link dokku.mongo.weso-staging:dokku-mongo-weso-staging
=====> weso-dashboard-backend-staging domains information
       Domains app enabled:           true
       Domains app vhosts:            upgrade-api-staging.weso-dashboard.com api-staging.weso-dashboard.com upgrade-static-staging.weso-dashboard.com static-staging.weso-dashboard.com
       Domains global enabled:        true
       Domains global vhosts:         weso-dashboard-upgrade-1659029142806-c-4-8gib-fra1-01
=====> weso-dashboard-backend-staging git information
       Git deploy branch:             master
       Git global deploy branch:      master
       Git keep git dir:              false
       Git rev env var:               GIT_REV
       Git sha:                       eb8d73f
       Git source image:
       Git last updated at:           1678919406
=====> weso-dashboard-backend-staging haproxy information
       Haproxy image:                 byjg/easy-haproxy:4.3.0
       Haproxy letsencrypt email:
       Haproxy letsencrypt server:    https://acme-v02.api.letsencrypt.org/directory
       Haproxy log level:             ERROR
=====> weso-dashboard-backend-staging logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:
       Logs max size:
       Logs vector sink:
=====> weso-dashboard-backend-staging network information
       Network attach post create:
       Network attach post deploy:
       Network bind all interfaces:          false
       Network computed attach post create:
       Network computed attach post deploy:
       Network computed bind all interfaces: false
       Network computed initial network:
       Network computed tld:
       Network global attach post create:
       Network global attach post deploy:
       Network global bind all interfaces:   false
       Network global initial network:
       Network global tld:
       Network initial network:
       Network static web listener:
       Network tld:
       Network web listeners:                172.17.0.13:5000
=====> weso-dashboard-backend-staging nginx information
       Nginx access log format:
       Nginx access log path:         /var/log/nginx/weso-dashboard-backend-staging-access.log
       Nginx bind address ipv4:
       Nginx bind address ipv6:       ::
       Nginx client max body size:
       Nginx disable custom config:   false
       Nginx error log path:          /var/log/nginx/weso-dashboard-backend-staging-error.log
       Nginx global hsts:             true
       Nginx computed hsts:           true
       Nginx hsts:
       Nginx hsts include subdomains: true
       Nginx hsts max age:            15724800
       Nginx hsts preload:            false
       Nginx computed nginx conf sigil path: nginx.conf.sigil
       Nginx global nginx conf sigil path: nginx.conf.sigil
       Nginx nginx conf sigil path:
       Nginx proxy buffer size:       4096
       Nginx proxy buffering:         on
       Nginx proxy buffers:           8 4096
       Nginx proxy busy buffers size: 8192
       Nginx proxy read timeout:      60s
       Nginx last visited at:         1704272936
       Nginx x forwarded for value:   $remote_addr
       Nginx x forwarded port value:  $server_port
       Nginx x forwarded proto value: $scheme
       Nginx x forwarded ssl:
=====> weso-dashboard-backend-staging openresty information
       Openresty access log format:
       Openresty access log path:     /var/log/nginx/weso-dashboard-backend-staging-access.log
       Openresty bind address ipv4:
       Openresty bind address ipv6:   ::
       Openresty client max body size:
       Openresty error log path:      /var/log/nginx/weso-dashboard-backend-staging-error.log
       Openresty global hsts:         true
       Openresty computed hsts:       true
       Openresty hsts:
       Openresty hsts include subdomains: true
       Openresty hsts max age:        15724800
       Openresty hsts preload:        false
       Openresty image:               dokku/openresty-docker-proxy:0.6.0
       Openresty letsencrypt email:
       Openresty letsencrypt server:  https://acme-v02.api.letsencrypt.org/directory
       Openresty proxy buffer size:   4096
       Openresty proxy buffering:     on
       Openresty proxy buffers:       8 4096
       Openresty proxy busy buffers size: 8192
       Openresty proxy read timeout:  60s
       Openresty x forwarded for value: $remote_addr
       Openresty x forwarded port value: $server_port
       Openresty x forwarded proto value: $scheme
       Openresty x forwarded ssl:
=====> weso-dashboard-backend-staging ports information
       Ports map:                     http:80:5000
       Ports map detected:            http:80:5000
=====> weso-dashboard-backend-staging proxy information
       Proxy enabled:                 true
       Proxy type:                    nginx
Error: No such object: 8dd12c144ebaaddb90a543946bff25a5f6de87fa3183b65c3e260ab31559c84d
Error: No such object: ea34da518ba4faccba6295a1419b1ac185687504cdaea6d37759bfa2dd5ff230
=====> weso-dashboard-backend-staging ps information
       Deployed:                      true
       Processes:                     2
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:
       Ps restart policy:             on-failure:10
       Restore:                       false
       Running:                       false
       Status queue 1:                missing (CID: 8dd12c144eb)
       Status web 1:                  missing (CID: ea34da518ba)
=====> weso-dashboard-backend-staging registry information
       Registry computed image repo:      dokku/weso-dashboard-backend-staging
       Registry computed push on release: false
       Registry computed server:
       Registry global push on release:
       Registry global server:
       Registry image repo:
       Registry push on release:
       Registry server:
       Registry tag version:
=====> weso-dashboard-backend-staging resource information
=====> weso-dashboard-backend-staging scheduler information
       Scheduler computed selected:   docker-local
       Scheduler global selected:     docker-local
       Scheduler selected:
=====> weso-dashboard-backend-staging scheduler-docker-local information
       Scheduler docker local disable chown:
       Scheduler docker local init process: true
       Scheduler docker local parallel schedule count:
=====> weso-dashboard-backend-staging storage information
       Storage build mounts:
       Storage deploy mounts:
       Storage run mounts:
=====> weso-dashboard-backend-staging traefik information
       Traefik api enabled:           false
       Traefik api vhost:             traefik.dokku.me
       Traefik basic auth password:
       Traefik basic auth username:
       Traefik dashboard enabled:     false
       Traefik image:                 traefik:v2.10
       Traefik letsencrypt email:
       Traefik letsencrypt server:    https://acme-v02.api.letsencrypt.org/directory
       Traefik log level:             ERROR
       Traefik priority:

```

### Additional information

I'm using the following buildpacks via .buildpacks file:

```
https://github.com/adam187/flynn-yarn-cache-fix.git
https://github.com/heroku/heroku-buildpack-nodejs.git
https://github.com/jontewks/puppeteer-heroku-buildpack.git
```

### Output of failing commands after running: dokku trace:off

```shell
-----> Building weso-dashboard-backend-staging from herokuish
 !     Deprecated: please upgrade plugin to use 'pre-build' plugin trigger instead of pre-build-buildpack
-----> Adding app specific deployment-keys to build environment ...
-----> Adding host-keys to build environment ...
       Adding shared keys
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Adding shared deployment-keys to build environment ...
Unable to find image 'dokku/herokuish:latest' locally
docker: Error response from daemon: pull access denied for dokku/herokuish, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
 !     Removing invalid image tag dokku/weso-dashboard-backend-staging:latest
 !     App build failed
 !     Error running rebuild against app weso-dashboard-backend-staging: exit status 1
```

### Output of failing commands after running: dokku trace:on

[See output here](https://gist.github.com/justintemps/e3fa76c9ae7b3195965d6d9d704e5a1b)

**Comment by kevinob11** [comment]
I'm having the same issue after updating to 0.32.3 and can't seem to figure out what is going on. I had to revert to a snapshot before the upgrade. Any luck getting this sorted?

We use a few non-core plugins, but none seemed to have updates:
- clickhouse
- hostkeys
- mysql
- postgres
- letsencrypt
- redis
- deployment_keys

**Comment by josegonzalez** [comment]
Upgrade both the [deployment-keys](https://github.com/cedricziel/dokku-deployment-keys) and [hostkeys](https://github.com/cedricziel/dokku-hostkeys-plugin) plugins. Both were incorrectly written previously and have been fixed as of their 0.3.0 releases.

**Comment by kevinob11** [comment]
@josegonzalez thank you! I'll test it soon (might be a bit, need to get a service window for a few apps) and will report back here.
